File: python/orders_print.py
from amqplib import client_0_8 as amqp
# Set up a connection and comms channel
conn = amqp.Connection(host="queue.eve-metrics.com:5672 ", userid="guest", password="guest", virtual_host="/", insist=False)
chan = conn.channel()
# Declare a queue and bind it to the order upload exchange
chan.queue_declare(queue="my_app_name", durable=False, exclusive=False, auto_delete=True)
chan.queue_bind(queue="my_app_name", exchange="market_order_uploads", routing_key="")
# Define a callback to receive the message
def recv_callback(msg):
  print "Received: " + msg.body
# Bind the consumer and callback
chan.basic_consume(queue="my_app_name", no_ack=True, callback=recv_callback, consumer_tag="testtag")
# Keep this thread open
while True:
  chan.wait()
chan.basic_cancel("testtag")

File: ruby/orders_print.rb
# Require amqp library
require 'rubygems'
require 'amqp'
require 'mq'
# Handle signals for clean shutdown on ctrl+c
Signal.trap('INT') { AMQP.stop{ EM.stop } }
Signal.trap('TERM'){ AMQP.stop{ EM.stop } }
# Start connection
AMQP.start(:host => 'queue.eve-metrics.com') do
  puts "Connected to AMQP server OK, listening for new messages"
  amq = MQ.new
  # Define queue and bind it to the fanout exchange for market order uploads
  amq.queue('CHANGEME_APP_NAME_GOES_HERE', :exclusive=>true, :auto_delete=>true).bind(amq.fanout('market_order_uploads')).subscribe do |msg|
    puts msg
  end
end
